airbyte-integrations/connectors/source-amazon-seller-partner/unit_tests/integration/test_report_based_streams.py

```diff
@@ -0,0 +1,266 @@
+                http_mocker._matchers.append(document_request_matcher)
+
+                http_mocker._mocker.get(
+                    requests_mock.ANY,
+                    additional_matcher=http_mocker._matches_wrapper(document_request_matcher),
+                    response_list=[{"content": document_response.body, "status_code": document_response.status_code}],
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_report_access_forbidden_when_read_then_no_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response(status_code=403))
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_given_report_status_cancelled_when_read_then_no_records(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(
+                    _check_report_status_request(_REPORT_ID).build(),
+                    _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.cancelled)
+                )
+
+                output = self._read(stream_name, config())
+                assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_given_report_status_fatal_when_read_then_exception_raised(self, http_mocker: HttpMocker) -> None:
+        for stream_name in STREAMS.keys():
+            with self.subTest(msg=stream_name):
+                _mock_auth(http_mocker)
+
+                http_mocker.post(_create_report_request(stream_name).build(), _create_report_response())
+                http_mocker.get(
+                    _check_report_status_request(_REPORT_ID).build(),
+                    _check_report_status_response(stream_name, processing_status=ReportProcessingStatus.fatal)
+                )
+
+                output = self._read(stream_name, config(), expecting_exception=True)
+                assert output.errors[-1].trace.error.failure_type == FailureType.system_error
+                assert output.errors[-1].trace.error.message == f"The report for stream '{stream_name}' was not created - skip reading"
+
+    @HttpMocker()
+    def test_given_report_with_incorrect_date_format_when_read_then_formatted(self, http_mocker: HttpMocker) -> None:
+        for stream_name, params in STREAMS.items():
+            if "date_field" not in params or "expected_date_value" not in params:
```

Comment:
It's a shame we can't use `skip` with `subTest`

Reply:
Now there's just a separate set of parameters